mars/tensor/indexing/fill_diagonal.py

```diff
@@ -184,10 +184,8 @@ def _tile_nd(cls, op, val):
         # if more than 3d, we will rechunk the tensor into square chunk
         # on the diagonal direction
         in_tensor = op.input
-        nsplits = np.array(in_tensor.nsplits)
-        if not np.issubdtype(nsplits.dtype, np.integer) or not np.all(
-            np.diff(nsplits, axis=1) == 0
-        ):
+        nsplits = [tuple(np.array(split)) for split in in_tensor.nsplits]
+        if set(nsplits) != 1:
```

Comment:
Should this be `len(set(nsplits)) != 1` ?

Reply:
Fixed.